osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTileProviderBasic.java

```diff
@@ -203,15 +203,18 @@ protected boolean isDowngradedMode(final long pMapTileIndex) {
 	 * @since 6.0.3
 	 * cf. https://github.com/osmdroid/osmdroid/issues/1172
 	 */
-	public static MapTileFileStorageProviderBase getMapTileFileStorageProviderBase(
+	protected MapTileFileStorageProviderBase getMapTileFileStorageProviderBase(
 			final IRegisterReceiver pRegisterReceiver,
 			final ITileSource pTileSource,
 			final IFilesystemCache pTileWriter
 	) {
 		if (pTileWriter instanceof TileWriter) {
-			return new MapTileFilesystemProvider(pRegisterReceiver, pTileSource);
+			return new MapTileFilesystemProvider(pRegisterReceiver, pTileSource, (TileWriter) pTileWriter);
+		} else if (pTileWriter instanceof SqlTileWriter) {
+			return new MapTileSqlCacheProvider(pRegisterReceiver, pTileSource, (SqlTileWriter) pTileWriter);
+		} else {
+			return new MapTileSqlCacheProvider(pRegisterReceiver, pTileSource);
```

Comment:
Very strange.

Reply:
The problem with the original method is that it does not inject the passed `IFilesystemCache pTileWriter` parameter to the created `MapTileSqlCacheProvider` instances. Therefore, even if I create my own CustomTileWriter and pass it, the corresponding `CacheProvider` won't use it -> the method is not doing what it is intended to do.